### PR TITLE
Add detailed pokemon info

### LINKED ISSUE #4: [add] new datasets - pokemon

### PR DESCRIPTION
**Pull request by hinsxd**
Close #4
 
Pulled data from https://www.kaggle.com/rounakbanik/pokemon.
Parsed types so that users can read numbers directly, rather than "quoted numbers" like '1' from csv.
Provided detailed nullable type for pokemon

### CODE REVIEW

**Comment by hinsxd** [comment]
I am wondering if we should put the data in a separate file so that the functions can be viewed easily.

**Comment by sarathm09** [comment]
Good job @hinsxd  😄

**Comment by sarathm09** [comment]
> I am wondering if we should put the data in a separate file so that the functions can be viewed easily.

Yes, I feel we should split it. It'll be cleaner and more maintainable.

**Comment by sarathm09** [comment]
Two things,

1. I feel that some keys do not add much value when used as a reusable data generation library. for example, _against_*,base_egg_steps,base_happiness,base_total,capture_rate,classfication,experience_growth,height_m,japanese_name,percentage_male,sp_attack,sp_defense_. This might add to the size of the library if it is not important fields. It's just my opinion, I'm okay with having these fields also. 😄 
2. Since we are using JS objects, we can probably use camelCase for the keys, so that the end-user doesn't feel confused with two types of naming conventions. Even I had added underscores initially, but later I realised that it feels a bit odd, so I'm planning to change it in all the files.

**Comment by hinsxd** [comment]
Good point. I removed some game-related status and keep the followings:
```
export type Pokemon = {
  abilities: string[];
  attack: number;
  classfication: string;
  defense: number;
  hp: number;
  japaneseName: string;
  name: string;
  id: number;
  generation: number;
  isLegendary: boolean;
  type: string[];
};

```
I choose to keep `japaneseName` because I guess there are people interested in the original name too. What do you think?

**Comment by sarathm09** [comment]
> Good point. I removed some game-related status and keep the followings:
> ```
> export type Pokemon = {
>   abilities: string[];
>   attack: number;
>   classfication: string;
>   defense: number;
>   hp: number;
>   japaneseName: string;
>   name: string;
>   id: number;
>   generation: number;
>   isLegendary: boolean;
>   type: string[];
> };
> 
> ```
> I choose to keep `japaneseName` because I guess there are people interested in the original name too. What do you think?

Yep, agreed 👍🏽

File: src/datasets/pokemonData.ts [hunk 1]
```diff
@@ -0,0 +1,10520 @@
...
+  attack: number;
+  classfication: string;
+  defense: number;
+  hp: number;
+  japaneseName: string;
+  name: string;
+  id: number;
+  generation: number;
+  isLegendary: boolean;
+  type: string[];
```
Comment (sarathm09) [inline src/datasets/pokemonData.ts]: Just a small change.
Shouldn't this be `types` as this is an array?

**Review by sarathm09 — APPROVED** [review]
I have a minor comment, but the current one looks fine too.

File: src/datasets/pokemonData.ts (hunk 1 above)
Comment (hinsxd) [reply src/datasets/pokemonData.ts]: Oops thats a typo :P Fixed!

**Comment by sarathm09** [comment]
Thank you for your contribution, @hinsxd 👍🏼 😄

**Comment by hinsxd** [comment]
No problem ;)

**Pull request merged**